data_structure/segment_trees.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2020 @author Lasha Bukhnikashvili
+ * @file
+ * Program describes usage of a popular Data Structure: Segment Trees
+ * in Finding maximum number and its index on a given Interval
+ * in logarithmic = Log(N) Time using SegmentTree Data Structures
+ * with PreProcessing O(N * Log(N)) Time
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+/**
+ * Each Node contains corresponding value
+ * in array and it's index
+ */
+struct Node {
+    int val;
+    int ind;
+} null;
```

Comment:
1. using global variables should be avoided as much as possible unless absolutely necessary. Pass variables as function arguments
2. short ambiguous global variable names like `null` will lead to alerts from LGTM checks